crates/nu-glob/src/lib.rs

```diff
@@ -983,7 +1011,13 @@ fn fill_todo(
                     //             .starts_with('.')
                     //     });
                     // }
-                    children.sort_by(|p1, p2| p2.file_name().cmp(&p1.file_name()));
+                    children.sort_by(|p1, p2| {
+                        if signals.interrupted() {
+                            std::cmp::Ordering::Equal
```

Comment:
Can this `Ordering::Equal` cause any weirdness? 

I think we should check below again for `signals.interrupted` and explicitly return the error.

Reply:
While I understand the potential for speedup, I think [`sort_by` may choose to panic](https://doc.rust-lang.org/std/primitive.slice.html#panics-38) if the interrupt changes the sort order mid-operation.

Changing a relation of three elements `a` `b` `c` from some ordering to `Equals` can violate both the [transitivity and duality properties for `PartialOrd`](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) and the [rules for `Ord`](https://doc.rust-lang.org/std/cmp/trait.Ord.html#corollaries)